src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java

```diff
@@ -107,6 +112,17 @@ public boolean createIfNotExists(Execution execution) {
 		);
 	}
 
+	public List<Execution> getScheduled(int limit) {
```

Comment:
I have been thinking... :)

Considering that we are including the data (I assume?), maybe it is better to have a streaming interface, something like
`getScheduled(Instant now, Consumer<Execution> consumer)`

The method would then call the consumer for every row in the result-set, so that it would not have to read everything into memory first. That would be a "client" decision, he could if he wanted to and new that is was safe memory-wise.

Reply:
Fair enough, seems like a good use for `Consumer<>`, will take this along :).